util/topsql/reporter/reporter.go

```diff
@@ -290,10 +299,30 @@ func (tsr *RemoteTopSQLReporter) doCollect(
 // takeDataAndSendToReportChan takes out (resets) collected data. These data will be send to a report channel
 // for reporting later.
 func (tsr *RemoteTopSQLReporter) takeDataAndSendToReportChan(collectedDataPtr *map[string]*dataPoints) {
+	// Fetch TopN dataPoints.
+	records := make([]*dataPoints, 0, len(*collectedDataPtr))
+	for _, v := range *collectedDataPtr {
+		records = append(records, v)
+	}
+	var evicted []*dataPoints
+	records, evicted = tsr.getTopNDataPoints(records)
```

Comment:
I think it is fine to not to evict it here, since immediately after that the map will be dropped entirely. Maybe we can simply ignore them at the report time?

Reply:
This evict is use to avoid report useless SQL/Plan. And since the  data structure of `evicted ` is a slice, it's  a little bit of trouble for ignore them at the report time, since we should traverse the slice for each SQL/Plan record.

Reply:
How about traversing the records and generate a "retain map" first, then iterate all records in the hash map and send records that only in the retain map?